src/mlpack/bindings/R/mlpack/R/inline.R

```diff
@@ -0,0 +1,17 @@
+inlineCxxPlugin <- function(...) {
+    is_macos <- Sys.info()[["sysname"]] == "Darwin"
+    openmp_flag <- if (is_macos) "" else "$(SHLIB_OPENMP_CFLAGS)"
+    plugin <-
+        Rcpp::Rcpp.plugin.maker(
+            include.before = "#include <mlpack.h>",
+            libs           = paste(openmp_flag, "$(LAPACK_LIBS) $(BLAS_LIBS) $(FLIBS)"),
+            LinkingTo      = c("RcppArmadillo", "Rcpp", "RcppEnsmallen", "mlpack"),
+            package        = "mlpack"
+        )
+    settings <- plugin()
+    settings$env$PKG_CPPFLAGS <- paste("-I../inst/include", openmp_flag)
+    # C++14 is required
+    settings$env$USE_CXX14 <- "yes"
```

Comment:
Not sure we need this as C++14 is a given since R 4.2.0 and R 4.4.0 will impose C++17 (details, which I always forget, are in WRE).

So how about if we prefix this with a test?

```suggestion
    if (getRversion() < "4.2.0") settings$env$USE_CXX14 <- "yes"
```